patchwork-gui/src/main/java/net/patchworkmc/mixin/gui/MixinAbstractButtonWidget.java

```diff
@@ -30,11 +32,41 @@
 	@Shadow
 	protected int height;
 
+	@Shadow
+	public boolean active;
+
+	@Shadow
+	public abstract boolean isHovered();
+
+	protected int packedFGColor = 0;
+
 	public int getHeight() {
 		return this.height;
 	}
 
 	public void setHeight(int value) {
 		this.height = value;
 	}
+
+	@ModifyVariable(method = "renderButton", at = @At(value = "INVOKE_ASSIGN"), index = 7)
+	public int hookRenderButton(int original) {
+		return getFGColor();
+	}
+
+	public int getFGColor() {
```

Comment:
Technically this should have a duck interface, but don't worry about it--there's a lot of jank going on with this module anyways that will be cleaned up when 1.15 comes around.